### PR TITLE
[casr-afl][casr-libfuzzer] Add hints

### PR DESCRIPTION
**Pull request by hkctkuy**

### CODE REVIEW

File: casr/src/bin/casr-libfuzzer.rs [hunk 1]
```diff
@@ -102,6 +102,15 @@ fn main() -> Result<()> {
...
                 .help("Add \"--casr-gdb-args \'./gdb_fuzz_target <arguments>\'\" to generate additional crash reports with casr-gdb (e.g., test whether program crashes without sanitizers)"),
         )
+        .arg(
+            Arg::new("hint")
+                .long("hint")
+                .value_name("HINT")
+                .action(ArgAction::Set)
+                .default_value("auto")
+                .value_parser(["auto", "gdb", "java", "js", "python", "san"])
+                .help("Hint to force analysis tool")
```
Comment (SweetVishnya) [inline casr/src/bin/casr-libfuzzer.rs:112]: Maybe, help message could be little bit more clear like

Force run of casr-HINT to analyze crashes

File: casr/src/bin/casr-libfuzzer.rs [hunk 2]
```diff
@@ -132,22 +141,30 @@ fn main() -> Result<()> {
...
         envs.insert("LD_PRELOAD".to_string(), util::get_atheris_lib()?);
         "casr-python"
-    } else if argv[0].ends_with("jazzer") || argv[0].ends_with("java") {
+    } else if hint == "java"
+        || hint == "auto" && (argv[0].ends_with("jazzer") || argv[0].ends_with("java"))
+    {
         "casr-java"
-    } else if argv[0].ends_with(".js")
+    } else if hint == "js"
+        || hint == "auto" && argv[0].ends_with(".js")
```
[the remark this answers is not in the document]
Comment (hkctkuy) [reply casr/src/bin/casr-libfuzzer.rs:157]: No, we don't

[the remark this answers is not in the document]
Comment (hkctkuy) [reply casr/src/bin/casr-libfuzzer.rs:157]: No, there's 2 options when we choose some tool:
- It's specified explicitly with hint
- There's some sign of it's tool and hint is `auto` (cuz if it's not `auto` we need to force some other tool)

[the remark this answers is not in the document]
Comment (hkctkuy) [reply casr/src/bin/casr-libfuzzer.rs]: Done

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [![Crates.io](https://img.shields.io/crates/v/casr)](https://crates.io/crates/casr)
 [![Documentation](https://docs.rs/libcasr/badge.svg)](https://docs.rs/libcasr)
-[![codecov](https://codecov.io/github/ispras/casr/graph/badge.svg?token=D9VY1WRWA7)](https://codecov.io/github/ispras/casr)
```
[the remark this answers is not in the document]
Comment (hkctkuy) [reply README.md:3]: Fixed

**Review by SweetVishnya — APPROVED** [review]
LGTM

**Review by Avgor46 — APPROVED** [review]

**Review by anfedotoff — APPROVED** [review]

**Pull request merged**